src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -1654,6 +1654,26 @@ CheckInstalledVersion(int elevel)
 }
 
 
+/*
+ * InstalledAndAvailableVersionsSame compares extension's available version and
+ * its current version from the pg_extension catalog table. If they are not same
+ * returns false, otherwise returns true.
+ */
+bool
+InstalledAndAvailableVersionsSame()
+{
+	char *installedVersion = InstalledExtensionVersion();
+	char *availableVersion = AvailableExtensionVersion();
+
+	if (strncmp(installedVersion, availableVersion, NAMEDATALEN) == 0)
```

Comment:
Small note: NAMEDATALEN was mostly because I copy-pasted some other strncmp, but it's probably always going to be big enough to compare versions.

Reply:
thinking in the same way